assets/docs/pages/operations/upgrade.md

```diff
@@ -0,0 +1,207 @@
+You can use this guide to upgrade the version of {{< reuse "/docs/snippets/kgateway.md" >}} for your control plane and data plane components, or to apply changes to the components' configuration settings.
+
+<!-- TODO upgrade guide when we have a minor version
+## Considerations
+Consider the following rules before you plan your kgateway upgrade.
+
+### Testing upgrades
+
+During the upgrade, pods that run the new version of the control plane and proxies are created. Then, the old pods are terminated. Because zero downtime is not guaranteed, try testing the upgrade in a staging environment before upgrading your production environment.
+
+### Patch and minor versions
+
+**Patch version upgrades**: </br>
+- You can skip patch versions within the same minor release. For example, you can upgrade from version {{< reuse "docs/versions/short.md" >}}.0 to {{< reuse "docs/versions/n-patch.md" >}} directly, and skip the patch versions in between.
+
+**Minor version upgrades**: </br>
+- Before you upgrade the minor version, always upgrade your _current_ minor version to the latest patch. This ensures that your current environment is up-to-date with any bug fixes or security patches before you begin the minor version upgrade process.
+- Always upgrade to the latest patch version of the target minor release. Do not upgrade to a lower patch version, such as {{< reuse "docs/versions/short.md" >}}.0, {{< reuse "docs/versions/short.md" >}}.1, and so on.
+- Do not skip minor versions during your upgrade. Upgrade minor release versions one at a time. 
+
+## Step 1: Prepare to upgrade
+
+1. **Minor version upgrades**: Before you upgrade to a new minor version, first upgrade your _current_ minor version to the latest patch.
+   1. Find the latest patch of your minor version by checking the [release changelog](https://github.com/kgateway-dev/kgateway/releases).
+   2. Follow this upgrade guide to upgrade to the latest patch for your current minor version.
+   3. Then, you can repeat the steps in this guide to upgrade to the latest patch of the next minor version.
+
+2. Check that your underlying infrastructure platform, such as Kubernetes, and other dependencies run supported versions for the kgateway version that you want to upgrade to.
+   1. Review the [supported versions](../../reference/versions/) for dependencies such as Kubernetes, Istio, Helm, and more.
```

Comment:
we are missing these versions in agw at the moment. also we mention istio here which I am not sure we should???